content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+---
+title: AWS OIDC Authentication
+description: Using OIDC to Authenticate SPIRE to AWS S3 on Kubernetes
+weight: 4
+toc: true
+aliases:
+menu:
+  spire:
+    weight: 50
+    parent: 'spire-try'
+---
+
+This tutorial builds on the [Kubernetes Quickstart](/spire/try/getting-started-k8s/) guide to describe how a SPIRE identified workload can, using a JWT-SVID, authenticate to Amazon AWS APIs, assume an AWS IAM role, and retrieve data from an AWS S3 bucket. This avoids the need to create and deploy AWS IAM credentials with the workload itself.
+
+In this tutorial you will learn how to:
+
+* Deploy the OIDC Discovery Provider Service
+* Create the required DNS A record to point to the Discovery Provider document
+* Create a sample AWS identity provider, policy, role, and S3 bucket
+* Test access to the S3 bucket
+
+## Prerequisites
+
+Note the following required accounts, prerequisites, and limitations before starting this tutorial:
+
+* You'll need access to the Kubernetes environment that you configured when going through [Kubernetes Quickstart](/spire/try/getting-started-k8s/) 
+* You'll need access to the AWS console to configure an identity provider, policy, role, and S3 bucket
+* This tutorial cannot run on Minikube as the Kubernetes environment must be network accessible by AWS
+* You'll need the ability to configure a DNS A record for the SPIRE OIDC Discovery Provider document location (see next section)
+
+### Required DNS A Record for the OIDC Discovery Provider IP Address
```

Comment:
I'm not sure why this section is here when it isn't possible to complete any of these steps until later, and you already call out that you need to complete the OIDC. Why do we ask the user to think about this now?

Reply:
My thinking is that it's something the user will have to think about what tool to use to set up the A record, so I put it front and center here. Also, I didn't want to interrupt the flow of the k8s commands with this big explanation.
That said, I'm fine with moving it into the main flow. I think if I move the client deployment step to the testing section, this DNS step can be the last of part 1 or maybe a new part 2.

Reply:
Yeah, your call, to me it's much more readable to just have a fairly short element in the pre-requisites that you'll need to be able to have a domain registered and the ability to set DNS A records associated with that domain, and move the rest into the main body.

Reply:
Done. I moved the DNS steps inline into a new part 2 and included an nslookup verification step.